query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,484 @@
+
+    return urls
+  }
+
+  // !parameters.rawAsset.isUrls && parameters.rawAsset.isUpload // asset is in storage
+
+  // prepare data object
+  const contentParameters: ContentParameters = parameters.rawAsset.asUpload
+  const dataObject = await prepareDataObject(contentParameters, parameters.blockNumber, parameters.contentOwner)
+
+  return dataObject
+}
+
+interface IExtractAssetParameters {
+  assetIndex: number
+  assets: NewAsset[]
+  db: DatabaseManager
+  blockNumber: number
+  contentOwner: typeof DataObjectOwner
+}
+
+/*
+  Selects asset from provided set of assets and prepares asset data fit to be saved to db.
+*/
+async function extractAsset(parameters: IExtractAssetParameters): Promise<AssetStorageOrUrls> {
+  // ensure asset index is valid
+  if (parameters.assetIndex > parameters.assets.length) {
+    return inconsistentState(`Non-existing asset extraction requested`, {
+      assetsProvided: parameters.assets.length,
+      assetIndex: parameters.assetIndex,
+    })
+  }
+
+  // convert asset to data object record
+  return convertAsset({
+    rawAsset: parameters.assets[parameters.assetIndex],
+    db: parameters.db,
+    blockNumber: parameters.blockNumber,
+    contentOwner: parameters.contentOwner,
+  })
+}
+
+/*
+  As a temporary messure to overcome yet-to-be-implemented features in Hydra, we are using redudant information
+  to describe asset state. This function introduces all redudant data needed to be saved to db.
+
+  Changes `result` argument!
+*/
+function integrateAsset<T>(propertyName: string, result: Object, asset: AssetStorageOrUrls) {
+  // helpers - property names
+  const nameUrl = propertyName + 'Urls'
+  const nameDataObject = propertyName + 'DataObject'
+  const nameAvailability = propertyName + 'Availability'
+
+  // is asset saved in storage?
+  if (!isAssetInStorage(asset)) {
+    // (un)set asset's properties
+    result[nameUrl] = asset
+    result[nameAvailability] = AssetAvailability.ACCEPTED
+    result[nameDataObject] = undefined // plan deletion (will have effect when saved to db)
+
+    return result
+  }
+
+  // prepare conversion table between liaison judgment and asset availability
+  const conversionTable = {
+    [LiaisonJudgement.ACCEPTED]: AssetAvailability.ACCEPTED,
+    [LiaisonJudgement.PENDING]: AssetAvailability.PENDING,
+  }
+
+  // (un)set asset's properties
+  result[nameUrl] = [] // plan deletion (will have effect when saved to db)
+  result[nameAvailability] = conversionTable[asset.liaisonJudgement]
+  result[nameDataObject] = asset
+}
+
+async function extractVideoSize(assets: NewAsset[], assetIndex: number | undefined): Promise<BN | undefined> {
+  // escape if no asset is required
+  if (assetIndex === undefined) {
+    return undefined
+  }
+
+  // ensure asset index is valid
+  if (assetIndex > assets.length) {
+    return inconsistentState(`Non-existing asset video size extraction requested`, {assetsProvided: assets.length, assetIndex})
```

Comment:
Here again the processor shouldn't halt when user makes a mistake in the metadata.

Reply:
Invalid state doesn't throw error anymore.